### ISSUE TITLE
Add Bazel as a Monorepo Tool

### ISSUE THREAD

**Issue by Trismaske** [michaelrambeau/bestofjs#612]
**Please briefly describe the project you'd like to add to _Best of JS_?**
Bazel is an open-source build and test tool similar to Make, Maven, and Gradle. It uses a human-readable, high-level build language. Bazel supports projects in multiple languages and builds outputs for multiple platforms. Bazel supports large codebases across multiple repositories, and large numbers of users.

**What is the URL of the project you'd like to add to _Best of JS_?**
https://bazel.build/

**Before submitting the request to add a new project, please check the following points:**

- [x] The project is related to the web platform or Node.js
- [x] The project has more than 100 stars on GitHub
- [x] The project has thorough documentation in English
- [x] The project is not deprecated and actively maintained

**How would this project be tagged? (Pick one or several existing tags from this list: https://bestofjs.org/tags)**
- [Building tool](https://bestofjs.org/projects?tags=build)
- [Monorepo](https://bestofjs.org/projects?tags=monorepo)
- [Testing framework](https://bestofjs.org/projects?tags=test-framework)

**Also you can suggest new tags to add if you think _Best of JS_ is missing useful tags.**

**Comment by michaelrambeau** [comment]
Hello @Trismaske thank you for the suggestion.

I know Bazel is a very popular tool (used by Google itself if I remember well), especially in the Java community, and I guess it can build anything but I wonder if it's relevant for _Best of JS_, if you would use it for JS/TS projects for example?

Or maybe you come from the State of JS survey and what we need is a bunch of extra projects not tracked by _Best of JS_?

Thank you and excuse for my ignorance!

**Comment by Trismaske** [comment]
Hi @michaelrambeau 

That's correct, I am from the State of JS survey and thought it would be cool to have Bazel as an option especially in terms of a build tool or for monorepo management.

We have found Bazel to be a cut above the rest in terms of managing a monorepo where there a number of different languages involved. For our specific use case, we have a NodeJS backend with clients in JS, Python, Java, Rust etc. and Bazel is really useful in terms of sharing code between the JS client & Server while doing the build for clients in other languages as well.

We found [this site](https://monorepo.tools/) to be a good starting point for comparing Bazel to other tools out there.

If you feel Bazel is not "JS" enough to be included, I would not argue as it's not commonly used when JS is the only language in the repo.

Cheers
